src/include/migraphx/instruction_ref.hpp

```diff
@@ -32,7 +32,40 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
 struct instruction;
+#if defined(_WIN32) && !defined(NDEBUG)
+class instruction_ref : public std::list<instruction>::iterator
+{
+public:
+    instruction_ref() = default;
+    instruction_ref(const std::list<instruction>::iterator& other)
```

Comment:
It seems it would be simpler to do `using std::list<instruction>::iterator::std::list<instruction>::iterator` here.